zone/zone.cpp

```diff
@@ -2907,4 +2904,76 @@ bool Zone::AllowManastoneClick()
 	return true;
 }
 
+uint32 Zone::GetZoneInstanceIDByCharacterAndZone(uint32 character_id, uint32 zone_id)
+{
+	std::pair<uint32, uint32> charzonePair = std::make_pair(character_id, zone_id);
+
+	auto itr = zone_character_instance_cache.find(charzonePair);
+
+	if (itr != zone_character_instance_cache.end())
+	{
+		if (itr->second.HasLockout(Timer::GetTimeSeconds()))
+		{
+			return itr->second.zone_instance_id;
+		}
+		else
+		{
+			itr = zone_character_instance_cache.erase(itr);
+		}
+	}
+	else
+	{
+		CharacterInstanceLockout zone_instance_lockout = database.GetZoneInstanceIDByCharacterID(character_id, zone_id);
+		zone_character_instance_cache[charzonePair] = zone_instance_lockout;
+
+		return zone_instance_lockout.zone_instance_id;
+	}
+
+	return GUILD_NONE;
```

Comment:
Should this be ZONE_NONE or something similar? Not sure if guild makes sense here since you're trying to return a zone instance id?

Reply:
There's parts of the legacy naming at play. It'll be a large, potentially breaking refactor. I'll consider it out of scope at first, we can gradually fix it over time.